### PR TITLE
feature: add support for scrapping thermal subsystem

### PR DESCRIPTION
**Pull request by rafal-krypa**
Thanks for creating this addon, it's a nice and much lighter alternative to [hassio-addons/addon-glances](https://github.com/hassio-addons/addon-glances).
Please consider adding support for crapping of thermal Linux subsystem devices which on many platforms contains information about CPU temperature.

### CODE REVIEW

**Comment by Sabuto** [comment]
Thanks for this, i can get the cpu temp and the hdd temp from the sensors input and smart devices input, does this provide a better way to get the sensors?

I'm glad you like the addon :)

**Comment by rafal-krypa** [comment]
Maybe I could have described it better. As a matter of fact it's a workaround for Raspberry Pi Linux kernel not reporting CPU temperature in hwmon interface, but still reporting it thermal subsystem. It's described in detail here: https://github.com/raspberrypi/linux/issues/2182.

In my specific case, when Hassio is running on RPI4, lm-sensors fail to see any sensors but CPU temperature can be read from `/sys/class/thermal/themal_zone0/temperature`.

**Comment by Sabuto** [comment]
Ahh ok that makes sense, I like this idea then and should hopefully fix #16, I will look and see if there is a way to only allow certain things on certain arch's

File: telegraf/run.sh [hunk 1]
```diff
@@ -143,6 +143,21 @@ if bashio::config.true 'ipmi_sensor.enabled'; then
...
+  do
+    bashio::log.info "...$i"
+    {
+      echo "[[inputs.file]]"
+      echo "  files = [\"$i/temp\"]"
+      echo "  name_override = \"`basename $i`\""
+      echo "  data_format = \"value\""
+      echo "  data_type = \"integer\""
+    } >> $CONFIG
+  done
```
Comment (Sabuto) [inline telegraf/run.sh]: could we wrap this in a check to see if the dir's exsist?

File: telegraf/run.sh [hunk 2]
```diff
@@ -143,6 +143,21 @@ if bashio::config.true 'ipmi_sensor.enabled'; then
   sed -i "s,TIMEOUT,${IPMI_TIMEOUT},g" $CONFIG
 fi
 
+if bashio::config.true 'thermal.enabled'; then
+  bashio::log.info "Updating config for thermal devices"
+  for i in /sys/class/thermal/thermal_zone*
```
Comment (Sabuto) [inline telegraf/run.sh]: above here put something like:

if bashio::fs.directory_exists '/sys/class/thermal/thermal_zone*'; then
  // Code here
else
  bashio::log.warn "Please disable this feature, it is not compatible with your system"
fi

something like that?

Comment (rafal-krypa) [reply telegraf/run.sh]: Good point, I will fix it. But this check is not enough. To have full verification, I would first have to check if directory `/sys/class/thermal/` exists and then if there are subdirectories named `thermal_zone*` in it.

On the other hand I can keep it short like:
```bash
  for i in `shopt -s nullglob; echo /sys/class/thermal/thermal_zone*`; do
  ...
```
which would guard against missing `/sys/class/thermal/` or missing `/sys/class/thermal/thermal_zone*`, but without an obvious place to produce a warning message if they don't exist.

Comment (Sabuto) [reply telegraf/run.sh]: so that command, essentially matches and then if there is a match run the code to add the inputs file?

Comment (rafal-krypa) [reply telegraf/run.sh]: Please check the updated code. It should now work correctly if there are thermal_zone sensors and if there are not. In the latter case the `for` loop just won't run.

**Pull request merged**